src/Orchard.Web/Modules/Orchard.MediaLibrary/Handlers/MediaPartHandler.cs

```diff
@@ -113,7 +116,13 @@ public class MediaPartHandler : ContentHandler {
 
         protected void RemoveMedia(MediaPart part) {
             if (!string.IsNullOrEmpty(part.FileName)) {
-                _mediaLibraryService.DeleteFile(part.FolderPath, part.FileName);
+                var mediaItemsUsingTheFile = _contentManager.Query<MediaPart, MediaPartRecord>()
+                                                            .ForVersion(VersionOptions.Latest)
```

Comment:
Thinking about your @sebastienros comment to query latest and published version here...
As I Know Medias are not draftable, so querying latest should be enough. The only case in which a Media can be in draft, right now, is after a clone/translation and it has not a published version. In all the other cases if a media is published it's the Latest also. If you agree I will keep this code as is.